plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java

```diff
@@ -0,0 +1,279 @@
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_PARTITION_STATISTICS;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_TABLE;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_TABLE_STATISTICS;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.UPDATE_PARTITION_STATISTICS;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.UPDATE_TABLE_STATISTICS;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+@Test(singleThreaded = true) // metastore invocation counters shares mutable state so can't be run from many threads simultaneously
+public class TestHiveMetastoreAccessOperations
+        extends AbstractTestQueryFramework
+{
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("hive")
+            .setSchema("test_schema")
+            .build();
+
+    private CountingAccessHiveMetastore metastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("hive").toFile();
+        metastore = new CountingAccessHiveMetastore(createTestingFileHiveMetastore(baseDir));
+
+        queryRunner.installPlugin(new TestingHivePlugin(metastore));
+        queryRunner.createCatalog("hive", "hive", ImmutableMap.of());
+
+        queryRunner.execute("CREATE SCHEMA test_schema");
+        return queryRunner;
+    }
+
+    @Test
+    public void testCreateTable()
+    {
+        assertMetastoreInvocations("CREATE TABLE test_create (id VARCHAR, age INT)",
+                ImmutableMultiset.builder()
+                        .add(CREATE_TABLE)
+                        .add(GET_DATABASE)
+                        .add(GET_TABLE)
+                        .add(UPDATE_TABLE_STATISTICS)
+                        .build());
+    }
+
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        assertMetastoreInvocations("CREATE TABLE test_ctas AS SELECT 1 AS age",
+                ImmutableMultiset.builder()
+                        .add(GET_DATABASE)
+                        .add(CREATE_TABLE)
+                        .add(GET_TABLE)
+                        .add(UPDATE_TABLE_STATISTICS)
+                        .build());
+    }
+
+    @Test
+    public void testSelect()
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_from",
+                ImmutableMultiset.builder()
+                        .add(GET_TABLE)
+                        .addCopies(GET_TABLE_STATISTICS, 1)
+                        .build());
+    }
+
```

Comment:
Could you please add a test for selecting from a partitioned table?

Reply:
Added.